### PR TITLE
Fixes create invite logic and adds many tests

### PR DESCRIPTION
**Pull request by drewstone**
## Description

Adds all relevant failing tests for the /createInvite route.

## Motivation and Context

We had 0 fail case tests.

## How Has This Been Tested?

Running the tests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have linted the code locally prior to submission.

### CODE REVIEW

File: server/routes/createInvite.ts [hunk 1]
```diff
@@ -13,43 +23,53 @@ const createInvite = async (models, req: UserRequest, res: Response, next: NextF
...
     });
-    if (!requesterIsAdminOrMod) return next(new Error('Must be an admin/mod to invite new members'));
+    if (requesterIsAdminOrMod.length === 0) return next(new Error(Errors.MustBeAdminOrMod));
   }
 
   const { invitedEmail } = req.body;
   if (req.body.invitedAddress) {
-    const existingAddress = await models.Address.findOne({ where: {
-      address: req.body.invitedAddress,
-      chain: req.body.invitedAddressChain,
```
Comment (drewstone) [inline server/routes/createInvite.ts:43]: @zakhap would be good to get your thoughts here and what was intended with this invitedAddressChain as it was not included anywhere in the existing tests you wrote.

**Comment by drewstone** [comment]
@zakhap I'm putting you in charge of the review here to help you understand what is required when writing tests. In writing the fail case tests, I was able to find bugs in your implementation of the `/createInvite` route. I'll be continuing onto more soon. Please review the tests added in `invites.spec.ts` and let me know if anything is confusing or needs clarification. There may also be some logic tweaks that I made that will need to be added back/changed in case it is inconsistent with your initial implementation.

File: server/routes/createInvite.ts [hunk 2]
```diff
@@ -4,6 +4,16 @@ import { Response, NextFunction } from 'express';
 import lookupCommunityIsVisibleToUser from '../util/lookupCommunityIsVisibleToUser';
 import { SERVER_URL, SENDGRID_API_KEY } from '../config';
 
+export const Errors = {
```
Comment (zakhap) [inline server/routes/createInvite.ts:7]: I really like this precedent to enumerate the Error types and then check for them in the tests.

File: server/routes/createInvite.ts (hunk 1 above)
Comment (zakhap) [reply server/routes/createInvite.ts:43]: I don't recall why the invitedAddressChain is included here as it's obviously not related to the check at hand.

**Pull request merged**